### ISSUE TITLE
Jakarta EE 10: Support Jakarta EE 10 Core Profile, Web Profile and Full Platform

### ISSUE THREAD

**Issue by jhanders34** [OpenLiberty/open-liberty#18722]
## Description of the high level feature, including any external spec links:  
Implement the Jakarta EE 10 Web Profile and Full Platform specifications.  

This feature includes creating of the Jakarta EE 10 webProfile-10.0, jakartaee-10.0 and jakartaeeClient-10.0 convenience features.  Additionally this feature is used to create the packages for users to download Jakarta EE 10 functionality.

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
## When complete & mandatory, add links to the UFO (Upcoming Feature Overview) document, FTS (Feature Test Summary), and blogs post issues(s):
- UFO: https://ibm.ent.box.com/file/1052182243536?s=wwazlekszkis9gpfpjttkz62n9m2zd87
- FTS: #24480 
- Beta Blog Post (if applicable): https://github.com/OpenLiberty/open-liberty/issues/23358
- Blog Post: #24758

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
## List of Steps to complete or get approvals / sign-offs for Onboarding to the Liberty release (GM date)

Instructions:
- Do the actions below and mark them complete in the checklist when they are done.
- Make sure all feature readiness approvers put the appropriate tag on the epic to indicate their approval.

## **Design**
Before Development Starts or 8 weeks before Onboarding
- [x] POC Design / UFO Review Scheduled (David Chang) or N/A.
- [x] POC Design / UFO Reviewed (Feature Owner) or N/A.
- [x] Complete any follow-ons from the POC Review.
- [x] Design / UFO Approval ([Alasdair Nottingham](https://github.com/NottyCode)) or N/A.
- [ ] No Design / No UFO Approval ([Alasdair Nottingham](https://github.com/NottyCode)) or N/A.
- [x] SVT Requirements identified. (Epic owner / Feature owner with SVT focal point)
- [x] ID Requirements identified ([Documenting Open Liberty](https://github.com/OpenLiberty/open-liberty/wiki/Documenting-Open-Liberty)). (Epic owner / Feature owner with ID focal point)
- [x] Create a child task of this epic entitled "Feature Test Summary" via [this template](https://github.com/OpenLiberty/open-liberty/issues/new?assignees=&labels=Feature+Test+Summary&template=feature_test_summary.md&title=). Add the link in above.

##
##

## **Development**
When active development has begun 
- [x] Add the "In Progress" label to this issue.

## **Beta**
If your feature, or portions of it, are going to be included in a beta  
Before Onboarding the beta
- [x] Beta Fence the functionality (`kind=beta`, `ibm:beta`, `ProductInfo.getBetaEdition()`)  

1 week before beta GA
- [x] Create, populate, and link to the [Beta blog post issue](https://github.com/OpenLiberty/open-liberty/issues/new?assignees=lauracowen%2C+jakub-pomykala&labels=&template=blog_post_beta.md&title=BETA+BLOG+-+title_of_your_update)

## **Legal**
3 weeks before Onboarding
- [x] Identify all open source libraries that are changing or are new. Work with Legal Release Services (Cass Tucker or Release PM) to get open source cleared and approved. Or N/A. (Epic Owner).   New or changed open source impacts license and Certificate of Originality.

## **Translation**
3 weeks before Onboarding
- [x] All new or changed PII messages are checked into the integration branch, before the last translation shipment out. (Epic Owner)

## **Feature Complete**
2 weeks before Onboarding
- [x] Implementation complete. (Epic owner / Feature owner)
- [x] All function tests complete. Ready for FAT Approval. (Epic owner / Feature owner)
- [ ] Review all known issues for Stop Ship. (Epic owner / Feature owner / PM)

## **Focal Point Approvals**
2 to 1 week before Onboarding
#### You **MUST** have the Design Approved or No Design Approved label before requesting focal point approvals.

All features (both "Design Approved" and "No Design Approved")
- [ ] **FAT** - ([OpenLiberty/fat-approvers](https://github.com/orgs/OpenLiberty/teams/fat-approvers)). SOE FATS are running successfully or N/A . Approver adds label focalApproved:fat to the Epic in Github.
- [x] **Demo** - ([Tom Evans](https://github.com/tevans78) or [Chuck Bridgham](https://github.com/cbridgha)). Demo is scheduled for an upcoming EOI. Approver adds label focalApproved:demo to the Epic in Github.
- [x] **Globalization** ([Sam Wong](https://github.com/samwatibm) - Liberty / Simy Cheeran - tWAS). Translation is complete or N/A. TVT - complete or N/A. Approver adds label focalApproved:globalization to the Epic in Github.

"Design Approved" features
- [x] **Accessibility** - ([Steven Zvonek](https://github.com/steven1046)). Accessibility testing is complete or N/A. Approver adds label focalApproved:accessibility to the Epic in Github.
- [x] **ID** - ([Karen Deen](https://github.com/chirp1)). Documentation work is complete or N/A . Approver adds label focalApproved:id to the Epic in Github.
- [x] **Performance** - ([Jared Anderson](https://github.com/jhanders34)). Performance testing is complete with no high severity defects or N/A . Approver adds label focalApproved:performance to the Epic in Github.
- [x] **Serviceability** - ([Don Bourne](https://github.com/donbourne)). Serviceability has been addressed.
- [ ] **STE** - (Swati Kasundra). STE chart deck is complete or N/A . Approver adds label focalApproved:ste to the Epic in Github.
- [x] **SVT** - ([Brian Hanczaryk](https://github.com/hanczaryk) - APS). SVT is complete or N/A . Approver adds label focalApproved:svt to the Epic in Github.

## **Ready for GA**
1 week before Onboarding
- [ ] No Stop Ship issues for the feature. (Epic owner / Feature owner / Release PM)
- [ ] Ship Readiness Review and Release Notes completed (Epic owner / Feature owner / Release PM)
- [ ] Github Epic and Epic's issues are closed / complete. All PRs are committed to the release branch. (Epic owner / Feature owner / Backlog Subtribe PM)

1 week before GA
- [ ] Create, populate, and link to the [Blog post issue](https://github.com/OpenLiberty/open-liberty/issues/new?assignees=lauracowen%2C+jakub-pomykala&labels=&template=blog_post_ga_release.md&title=GA+BLOG+-+title_of_your_update)

## **Other deliverbles**
- [ ] **OL Guides** - ([Yee-Kang Chang](https://github.com/yeekangc)). Assessment for OL Guides is complete or N/A.
- [ ] **WDT** - (Leonard Theivendra). WDT work complete or N/A.
- [ ] **Blog** - ([Laura Cowen](https://github.com/lauracowen)) Blog article writeup (Epic owner / Feature owner / Laura Cowen)

**Comment by donbourne** [comment]
Notes from UFO socialization (see required actions marked with ***):

**p. 8**
    *** fix text color **(fixed)**
    
**p. 10**
    why is cdi lite a requirement for core profile?  core profile includes cdi lite
    nobody would complain if we included cdi full
    use cdi lite to satisfy that
    emily - think we could have a coreProfile-10.0 (restfulWS (CDI/JSON-P) and JSON-B (...))
        people use coreProfile for their apps    
        alasdair - I don't really agree - if we do core profile without cdi-lite 4.0 that doesn't make sense
        jared - have a cdi lite that is same as cdi full, but exposes subset of api?
            alasdair - would have to subset the pkg
        for cdi lite:
            - subset api - no dedicated api jar for that
            - session scope not supported
            - beans.xml - if discovery mode = all should throw an error message
        emily - agree there should be no cdi-lite feature
            we use cdi-full to certify the whole profile
            having coreProfile convenience feature is ok
        *** emily/alasdair/jared to take this offline to discuss whether we should have coreProfile-10.0 feature
**Kept it the way it is documented.  No coreProfile-10.0 feature.  Added documentation on what features needed in server.xml to get core profile function.**
        
**p. 11**
    *** add the list of value-add features that are being updated **(added)**
    *** password utilities 1.1 also needs to be added **(included)**
    *** explain what repeat means in technical background section **(added)**
        
**p. 27**
    should webProfile archives include xmlBinding-4.0 liberty feature to match ee 9?
    emily - webprofile - name is strange (jakarta ee 10) 
        - same naming as used in jakarta ee 9
        - how many download for webProfile 10?
        - does jakarta ee include MP? yes
        - you can't include MP in webProfile 10 - initially will not include
        - for archive, only web profile features?  or everything?  jared: just what's required for web profile 10
        - when we first release 22.0.0.13 - to start with it won't include MP?  (of course, it's not GA yet)
        - when MP goes live (23.0.0.1) - will we redo this zip? historically, it did
            michal: i don't think we should include MP in this
            alasdair: we have in the past, added mp to jakarta ee ones, when mp ones become available.  if we want to discuss, have to discuss all of those zip files
                - broader discussion than just MP on this
            tom evans: hard to remove things later
            *** take this conversation/decision offline [alasdair / emily / jared / tom e / chuck / michal]
        - Chuck - low % of downloads are for the convenience packages. most are for kernel or full download
            alasdair - web profile gets a lot of downloads (looking at DHE)
            **Addressed at POC.  Will include MP features in archive.  Will keep the name the same.  Will NOT include xmlBinding-4.0 in webprofile 10 archive.  Will stop publishing EE 9 archives and have EE 10 instead.**
            
**p. 29**
    *** decision - do not have a message for use of deprecated security manager
        java security manager puts out message in v18

**Comment by chirp1** [comment]
Documentation issue at https://github.com/OpenLiberty/docs/issues/5969 Information to document received. Documentation in progress. Approving this epic.

**Comment by donbourne** [comment]
as this feature is about creating the umbrella features, no need for serviceability approval.

**Comment by jhanders34** [comment]
@OpenLiberty/ste-approvers are you wanting an STE for this Epic?
@OpenLiberty/accessibility-approvers do you need anything to approve this Epic?
@OpenLiberty/globalization-approvers Sam, all NLS has been translated for this Epic months ago so I think you can approve.

**Comment by tngiang73** [comment]
@jhanders34 : L2 has requested STE slides for this feature.  The STE template can be found at the links below.  You can use either one to create education.

Slide Template:  https://ibm.box.com/s/1an42g7zdgmaj84w7dft0indqfgi8ffm
Github Template:  https://pages.github.ibm.com/WASL3/site/STE/about

Please upload the completed slides to the same 'STE Archive' BOX folder (https://ibm.box.com/s/m5i0kgube9xyxgfckpi8gru69ow17zay) and let Thanh Giang ([@tngiang73 ](https://github.ibm.com/tgiang)73) or Geeta Nadella ([@gnadell](https://github.ibm.com/gnadell)) know they're complete or provide us the Github link. Once the completed STE slide is provided, we will provide the STE approval. Thanks!

**Comment by jhanders34** [comment]
@OpenLiberty/demo-approvers I have added demo to the March 28 EOI agenda.

**Comment by jhanders34** [comment]
@OpenLiberty/fat-approvers I have included a link to the FTS for this feature.

**Comment by tngiang73** [comment]
L2 is good with the STE slides. Thanks.
